code/modules/reagents/reagents/addictions.dm

```diff
@@ -0,0 +1,47 @@
+#define TICK_ADDICTION_REMINDER 50
+
+/datum/addiction
+	var/name = "addiction"
+	var/tick //Tick goes up if you're not satiating your addiction, Tick goes down if you're satiating it
+	var/addiction = "memes" //Holds the ID for whatever reagent you're addicted to
+	var/mob/holder
+	var/addiction_multiplier = 1
+	var/list/addiction_effects = list() //What effects this addiction will have on the holder. Organized as list(tick_to_trigger = effect)
+
+/datum/addiction/New(var/mob/new_holder, var/addicted)
+	..()
+	if(new_holder)
+		holder = new_holder
+	if(addicted)
+		addiction = addicted
+
+/datum/addiction/Destroy()
+	if(holder)
+		holder = null
+	if(addiction_effects)
+		for(var/datum/addiction_effect/E in addiction_effects)
+			addiction_effects.Remove(E)
```

Comment:
don't remove each item individually, null the list after you qdel'd the elements instead